src/utils/client.js

```diff
@@ -161,8 +167,24 @@ function getPropByString(obj, propString) {
   return obj[props[i]];
 }
 
+// compare fields in the echis client doc and the client doc we got from Client Registry
+function getMismatchedClientFields(echisClientDoc, crClientDoc) {
+  // fields that ideally should be kind of unique across clients
+  const matcherFields = ["firstName", "middleName", "lastName", "gender", "dateOfBirth", "contact.primaryPhone"];
+  // where we store our mismatched fields and return them later
+  const mismatchedFields = {};
+  // the payload we generate here has the same format as what we get back when we query for client existence
+  const crPayload = generateClientRegistryPayload(echisClientDoc);
+  matcherFields.forEach(key => {
+    if (getPropByString(crClientDoc, key).toUpperCase() !== getPropByString(crPayload, key).toUpperCase()) {
+      mismatchedFields[key] = { Have: crPayload[key], Expected: crClientDoc[key] };
```

Comment:
Any preference for `Have` and `Expected` to `have` and `expected`? `actual` vs `expected` sounds better - following common practice in _testing_.

Reply:
done